ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Stages/InferredTypeCodeVisionProvider.fs

```diff
@@ -0,0 +1,102 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Stages
+
+open FSharp.Compiler.SourceCodeServices
+open JetBrains.ReSharper.Daemon.CodeInsights
+open JetBrains.ReSharper.Feature.Services.Daemon
+open JetBrains.ReSharper.Plugins.FSharp.Daemon.Cs.Stages
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi
+open JetBrains.Rider.Model
+
+[<DaemonStage>]
+type InferredTypeCodeVisionProvider() =
+    inherit FSharpDaemonStageBase()
+
+    override x.CreateStageProcess(fsFile, settings, daemonProcess) =
+        InferredTypeCodeVisionProviderProcess(fsFile, settings, daemonProcess) :> _
+
+and InferredTypeCodeVisionProviderProcess(fsFile, settings,  daemonProcess) =
+    inherit FSharpDaemonStageProcessBase(fsFile, daemonProcess)
+
+    let [<Literal>] id = "Inferred types"
+
+    let formatMemberOrFunctionOrValue (mfv: FSharpMemberOrFunctionOrValue, symbolUse: FSharpSymbolUse) =
+        let returnTypeStr = mfv.ReturnParameter.Type.Format symbolUse.DisplayContext
+
+        let parameterGroups =
+            if mfv.IsPropertyGetterMethod then []
+            else
+              let groups =
+                [ for group in mfv.CurriedParameterGroups ->
+                    [ for p in group do
+                        let typeStr = p.Type.Format symbolUse.DisplayContext
+                        if p.Type.IsFunctionType then yield sprintf "(%s)" typeStr
+                        else yield typeStr ]]
+
+              match groups with
+              | [[]] when mfv.IsMember -> [["unit"]]
+              | _ -> groups
+
+        [ for group in parameterGroups do
+            let groupStr = group |> String.concat " * "
+            if group.Length = 1 then yield groupStr
+            else yield sprintf "(%s)" groupStr
+          yield returnTypeStr ]
+        |> String.concat " -> "
```

Comment:
Is it possible to add formatting here? Like colorizing the type names different from symbols ("like ->").

Also, font ligatures doesn't seem to work inside code vision. Perhaps the unicode arrow would be better here?

Last question, perhaps for @auduchinok : can you change the font/size/weight of code vision? It hurts my OCD that the font isn't monospaced xD

Reply:
This is how it looks like with a unicode arrow:

![image](https://user-images.githubusercontent.com/873919/61434469-b7e7d080-a93e-11e9-8434-c4c177382a14.png)